### PR TITLE
r2dec for r2pm

### PR DESCRIPTION
**Pull request by wargio**
script not tested yet. wait before merge.

### CODE REVIEW

File: db/r2dec [hunk 1]
```diff
@@ -0,0 +1,21 @@
+R2PM_BEGIN
+
+R2PM_GIT "https://github.com/wargio/r2dec-js"
+R2PM_DESC "[r2-r2pipe-node] an Experimental Decompiler"
+BINDIR="${R2PM_BINDIR}"
+
+R2PM_INSTALL() {
+	node --version || exit 1
+	npm install || exit 1
+	mkdir -p "${R2PM_BINDIR}" || exit 1
```
Comment (radare) [inline db/r2dec]: this dir should be already mkdir'd by r2pm, but i guess its fine to have it here

File: db/r2dec [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	echo '#!/bin/sh' > "${R2PM_BINDIR}/r2dec" || exit 1
+	echo "exec node '`pwd`'"'/r2dec.js $*' >> "${R2PM_BINDIR}/r2dec" || exit 1
```
Comment (radare) [inline db/r2dec]: why not just copy the r2dec.js with #!/usr/bin/env node as hashbang instead of relaying on /bin/sh to wrap that

Comment (radare) [inline db/r2dec]: you can even ln -fs to avoid the copy and ease the tests when changing the code

File: db/r2dec [hunk 3]
```diff
@@ -0,0 +1,21 @@
+R2PM_BEGIN
+
+R2PM_GIT "https://github.com/wargio/r2dec-js"
+R2PM_DESC "[r2-r2pipe-node] an Experimental Decompiler"
+BINDIR="${R2PM_BINDIR}"
+
+R2PM_INSTALL() {
+	node --version || exit 1
+	npm install || exit 1
```
Comment (radare) [inline db/r2dec]: i would go for removeing the package-lock.json too , that little bastard that fucks all our npm setups

File: db/r2dec [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	mkdir -p "${R2PM_BINDIR}" || exit 1
+	echo '#!/bin/sh' > "${R2PM_BINDIR}/r2dec" || exit 1
+	echo "exec node '`pwd`'"'/r2dec.js $*' >> "${R2PM_BINDIR}/r2dec" || exit 1
+	chmod +x "${R2PM_BINDIR}"/r2dec || exit 1
+}
+
+R2PM_UNINSTALL() {
+	rm -f "${BINDIR}"/r2dec
+	rm -rf "${R2PM_GITDIR}"/r2dec
```
Comment (radare) [inline db/r2dec]: nope dont remove this here, uninstall != purge that purke is done by r2pm -k and its implicity of the pkg system

File: db/r2dec (hunk 1 above)
Comment (wargio) [inline db/r2dec]: ok. i'll remote it

File: db/r2dec (hunk 2 above)
Comment (wargio) [inline db/r2dec]: hmmm maybe it's better to still install the deps and then do the cp

File: db/r2dec (hunk 3 above)
Comment (wargio) [inline db/r2dec]: agree

File: db/r2dec (hunk 4 above)
Comment (wargio) [inline db/r2dec]: ok

File: db/r2dec (hunk 2 above)
Comment (radare) [inline db/r2dec]: up2u

**Pull request merged**